### PR TITLE
Update AWS gem info content

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
Following AudioEngineWwise as an example

Screen Shot 2021-11-03 at 3 54 56 PM

Signed-off-by: liug <liug@amazon.com>

### CODE REVIEW

File: Gems/AWSClientAuth/gem.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
         "SDK"
     ],
     "icon_path": "preview.png",
-    "requirements": "",
```
Comment (lmbr-pip) [inline Gems/AWSClientAuth/gem.json:17]: nit: why remove this? Most gems.json have this set as an empty string.

File: Gems/AWSGameLift/gem.json [hunk 2]
```diff
@@ -11,10 +11,11 @@
     "user_tags": [
         "AWS",
         "Framework",
-        "Network"
+        "Network",
+        "SDK"
     ],
     "icon_path": "preview.png",
-    "requirements": "",
+    "requirements": "Users will need to enable Multiplayer Gem together to utilize AWSGameLift feature.",
```
Comment (lmbr-pip) [inline Gems/AWSGameLift/gem.json:18]: nit: "Users will need to enable the Multiplayer gem to support the AWSGameLift feature"

However, I don't see any other gem with the requirements filed set (https://github.com/o3de/o3de/search?l=JSON&p=11&q=requirements), is this expected to be a descriptive string?

File: Gems/AWSClientAuth/gem.json (hunk 1 above)
Comment (allisaurus) [reply Gems/AWSClientAuth/gem.json:17]: if this field is expected to be a description - should we list the AWSCore gem as a requirement for this and the AWSMetrics gem?

Comment (onecent1101) [reply Gems/AWSClientAuth/gem.json:17]: dependencies are defined in "dependencies" field.
remove this to keep file clean if we have nothing to provide in this field.

**Review by lmbr-pip — APPROVED** [review]
LG.

**Review by lmbr-pip — APPROVED** [review]

**Review by junbo75 — APPROVED** [review]

**Pull request merged**